bundles/binding/org.openhab.binding.powermax/src/main/java/org/openhab/binding/powermax/internal/PowerMaxBinding.java

```diff
@@ -0,0 +1,907 @@
+            connected = comm.open();
+            if (serialPort != null) {
+                logger.info("PowerMax alarm binding: serial connection ({}): {}", serialPort,
+                        connected ? "connected" : "disconnected");
+            } else if (ipAddress != null) {
+                logger.info("PowerMax alarm binding: TCP connection (IP {} port {}): {}", ipAddress, tcpPort,
+                        connected ? "connected" : "disconnected");
+            }
+        } else {
+            connected = false;
+        }
+        logger.debug("openConnection(): {}", connected ? "connected" : "disconnected");
+    }
+
+    /**
+     * Close TCP or Serial connection to the PowerMax Alarm Panel and remove the Event Listener
+     */
+    private void closeConnection() {
+        PowerMaxCommDriver comm = PowerMaxCommDriver.getTheCommDriver();
+        if (comm != null) {
+            comm.close();
+            comm.removeEventListener(this);
+        }
+        connected = false;
+        logger.debug("closeConnection(): disconnected");
+    }
+
+    /**
+     * PowerMax Alarm incoming message event handler
+     *
+     * @param event
+     */
+    @Override
+    public void powerMaxEventReceived(EventObject event) {
+        PowerMaxEvent powerMaxEvent = (PowerMaxEvent) event;
+        PowerMaxBaseMessage message = powerMaxEvent.getPowerMaxMessage();
+
+        logger.debug("powerMaxEventReceived(): received message {}", (message.getReceiveType() != null)
+                ? message.getReceiveType().getDescription() : String.format("%02X", message.getCode()));
+
+        if (message instanceof PowerMaxInfoMessage) {
+            ((PowerMaxInfoMessage) message).setAutoSyncTime(autoSyncTime);
+        }
+
+        PowerMaxState updateState = message.handleMessage();
+        if (updateState != null) {
+
+            if ((currentState.isPowerlinkMode() != null) && currentState.isPowerlinkMode().equals(Boolean.TRUE)
+                    && (updateState.isDownloadSetupRequired() != null)
+                    && updateState.isDownloadSetupRequired().equals(Boolean.TRUE)) {
+                // After Enrolling Powerlink or if a reset is required
+                logger.info("PowerMax alarm binding: Reset");
+                PowerMaxCommDriver.getTheCommDriver().startDownload();
+                if (currentState.getLastKeepAlive() != null) {
+                    currentState.setLastKeepAlive(System.currentTimeMillis());
+                }
+            } else if ((currentState.isPowerlinkMode() != null) && currentState.isPowerlinkMode().equals(Boolean.FALSE)
+                    && (updateState.getLastKeepAlive() != null)) {
+                // Were are in standard mode but received a keep alive message
+                // so we switch in PowerLink mode
+                logger.info("PowerMax alarm binding: Switching to Powerlink mode");
+                PowerMaxCommDriver.getTheCommDriver().startDownload();
```

Comment:
All my logs at INFO or WARN level start with "PowerMax alarm binding" while for DEBUG level, I start the message with the name of the method. The name of the method is not talking to the final user.